lib/src/ima.rs

```diff
@@ -47,10 +48,19 @@ fn xattrs_to_map(v: &glib::Variant) -> BTreeMap<Vec<u8>, Vec<u8>> {
     map
 }
 
-/// Reserialize a map to GVariant of type `a(ayay)`
-fn xattrmap_serialize(map: &BTreeMap<Vec<u8>, Vec<u8>>) -> glib::Variant {
-    let map: Vec<_> = map.iter().collect();
-    variant_utils::new_variant_a_ayay(&map)
+/// Create a new GVariant of type a(ayay).  This is used by OSTree's extended attributes.
+fn new_variant_a_ayay<'a, T: 'a + AsRef<[u8]>>(
+    items: impl IntoIterator<Item = (T, T)>,
+) -> glib::Variant {
+    let children: Vec<_> = items
+        .into_iter()
+        .map(|(a, b)| {
+            let a = a.as_ref();
+            let b = b.as_ref();
+            Variant::from_tuple(&[a.to_variant(), b.to_variant()])
+        })
+        .collect();
+    Variant::from_array::<(&[u8], &[u8])>(&children)
```

Comment:
Self-note: this internally gets ownership of the content, see note at https://developer.gnome.org/glib/stable/glib-GVariant.html#g-variant-new-array.